ExampleApp/ExampleApp/Screens/PeripheralServices/ViewController/PeripheralServicesViewController.swift

```diff
@@ -0,0 +1,152 @@
+import RxBluetoothKit
+import RxCocoa
+import RxSwift
+import UIKit
+
+final class PeripheralServicesViewController: UIViewController, CustomView {
+
+    typealias ViewClass = BaseView
+
+    typealias PeripheralServicesDataSource = TableViewDataSource<[Service], PeripheralServicesViewModelItem>
+
+    private let viewModel: PeripheralServicesViewModelType
+
+    private let dataSource: PeripheralServicesDataSource
+
+    private let disposeBag: DisposeBag = DisposeBag()
+
+    init(with dataSource: PeripheralServicesDataSource, viewModel: PeripheralServicesViewModelType) {
+        self.dataSource = dataSource
+        self.viewModel = viewModel
+        super.init(nibName: nil, bundle: nil)
+    }
+
+    @available(*, unavailable)
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func loadView() {
+        view = ViewClass()
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setCustomView()
+        setupTableView()
+        setDataSourceBlocks()
+        adjustTitle()
+        bindViewModel()
+    }
+
+    private func setCustomView() {
+        customView.toggleActivityIndicator(true)
+    }
+
+    private func setupTableView() {
+        customView.setTableView(dataSource: dataSource, delegate: self)
+        registerCells()
+    }
+
+    private func setDataSourceBlocks() {
+        setDataSourceRefreshBlock()
+        setDataSourceOnErrorBlock()
```

Comment:
again i don't think `registerCells` and DataSource setup needs to be extracted to separate functions. It overly extends class